apps/prairielearn/python/colors.py

```diff
@@ -1,196 +1,139 @@
+    "turquoise2": (39, 203, 192, 255),
+    "turquoise3": (0, 139, 128, 255),
+    "yellow1": (253, 227, 167, 255),
+    "yellow2": (245, 171, 53, 255),
+    "yellow3": (216, 116, 0, 255),
 }
 
+PL_COLORS_VALUE_MAP = dict([(v, k) for k, v in PL_COLORS_NAME_MAP.items()])
+
+
+class PrarieLearnColor(sRGB):
+    """Custom sRGB class to handle custom PrarieLearn colors, via Coloraide."""
+
+    def match(self, string, start=0, fullmatch=True):
+        """
+        Match a color string, first trying PrarieLearn.
+        If no match is found, defaults to sRGB class' implementation.
+        """
+        # Match the string using fullmatch if requested
+        match = (
+            RE_PL_COLORS.fullmatch(string, start)
+            if fullmatch
+            else RE_PL_COLORS.match(string, start)
+        )
+
+        if match:
+            # See if we can find the name
+            name = match.group(1)
+            values = PL_COLORS_NAME_MAP.get(name.lower(), None)
+
+            if values is not None:
+                # Normalize back to 0 - 1
+                values = [c / 255 for c in values]
+                # Return the coordinates and transparency separate,
+                # also include the match end position.
+                return (values[:-1], values[-1]), match.end(1)
+
+        # Couldn't find custom colors, so use the default color matching.
+        return super().match(string, start, fullmatch)
+
+    def to_string(
+        self,
+        parent,
+        *,
+        alpha=None,
+        precision=None,
+        fit=True,
+        none=False,
+        color=False,
+        hex=False,  # pylint: disable=redefined-builtin
+        names=False,
+        comma=False,
+        upper=False,
+        percent=False,
+        compress=False,
+        **kwargs,
+    ) -> str:
+        """Convert to string."""
+
+        if names:
+            # Get alpha and coordinates resolving undefined values as required
+            alpha = serialize.get_alpha(parent, alpha, False, False)
+            if alpha is None:
+                alpha = 1
+            coords = serialize.get_coords(parent, fit, False, False) + [alpha]
+
+            # See if the color value is a match, if so, return the string
+            result = PL_COLORS_VALUE_MAP.get(
+                tuple([alg.round_half_up(c * 255) for c in coords]), None
+            )
+            if result is not None:
+                return result
+
+        # Color name not requested, or there was no color match, use default serialization.
+        return super().to_string(
+            parent,
+            alpha=alpha,
+            precision=precision,
+            fit=fit,
+            none=none,
+            color=color,
+            hex=hex,
+            names=names,
+            comma=comma,
+            upper=upper,
+            percent=percent,
+            compress=compress,
+            **kwargs,
+        )
 
-def get_css_color(name: str) -> Optional[str]:
```

Comment:
I think we might have to keep this function for backward compatibility (since I think instructor code is allowed to call this file), but not 100%. Either way, much nicer to have a full-fledged class integrate all of this.

Reply:
Good catch; I'll add it back in. Thankfully we can simplify the implementation of `get_css_color` (while retaining the same functionality) via the class.